### PR TITLE
Development logic

### PR DESCRIPTION
**Pull request by agawda**
It is possible now to get a board filled with a given list of ships, however there is not ship generator yet, so you have to insert them manually. It is also possible to display the board. Also, the Ship class is present, but as I mentioned, you have to code the ships manually.

### CODE REVIEW

File: server/src/main/java/com/blackship/battlesheep/game/StartingGame.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    public StartingGame() {
+        this.playerOneBoard = new StartingBoard();
+    }
+    //TODO: these methods are not used, but I need them here
+    public StartingGame(Function<List<Ship>, Board> insertShips) {
+        this.insertShips = insertShips;
+    }
+
+    public void fillBoard(List<Ship> ships) {
```
Comment (mhbratek) [inline server/src/main/java/com/blackship/battlesheep/game/StartingGame.java:26]: Broken SRP Starting game Filling board? Pass board to BoardModifier or something like that. Start game from method ```start()```.

File: server/src/main/java/com/blackship/battlesheep/game/StartingGame.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * @author Anna Gawda
+ * @since 26.07.17
+ */
+public class StartingGame implements Game {
+    private Board playerOneBoard;
+    private Function<List<Ship>, Board> insertShips;
```
Comment (mhbratek) [inline server/src/main/java/com/blackship/battlesheep/game/StartingGame.java:16]: :+1:

File: server/src/main/java/com/blackship/battlesheep/game/state/board/StartingBoard.java [hunk 3]
```diff
@@ -1,42 +1,51 @@
...
+    }
+
     private boolean isNewLine(Integer x) {
         return x % ROW_SIZE == 0;
     }
 
     private void fillStartingPositions() {
         IntStream.rangeClosed(BOARD_FIRST_FIELD, BOARD_SIZE)
-                .forEach(positions::add);
+                .forEach(x -> positions.put(x, new Position(x)));
```
Comment (mhbratek) [inline server/src/main/java/com/blackship/battlesheep/game/state/board/StartingBoard.java:49]: What is x?

File: server/src/test/java/com/blackship/battlesheep/game/state/board/BoardModifierTest.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    @BeforeTest
+    public void setupBoard() {
+        board = new StartingBoard();
+    }
+
+    @Test(dataProvider = "boardSheepData")
+    public void insertShipExpectBoardWithInsertedShip(List<Integer> givenShip) {
+        //given StartingBoard
```
Comment (mhbratek) [inline server/src/test/java/com/blackship/battlesheep/game/state/board/BoardModifierTest.java:47]: I think is not necessary to add this comment, I know there is ```@BeforeTest```.

File: server/src/test/java/com/blackship/battlesheep/game/state/board/BoardModifierTest.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        };
+    }
+
+    @BeforeTest
+    public void setupBoard() {
+        board = new StartingBoard();
+    }
+
+    @Test(dataProvider = "boardSheepData")
+    public void insertShipExpectBoardWithInsertedShip(List<Integer> givenShip) {
```
Comment (mhbratek) [inline server/src/test/java/com/blackship/battlesheep/game/state/board/BoardModifierTest.java:46]: In the future change name convention.

File: server/src/test/java/com/blackship/battlesheep/game/state/board/BoardModifierTest.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        //given StartingBoard
+
+        //when
+        Board givenBoard  = BoardModifier.insertShip(board, givenShip);
+        //then
+        assertEquals(givenBoard.getPositions().get(givenShip.get(0)), new Position(givenShip.get(0), FieldState.TAKEN));
```
Comment (mhbratek) [inline server/src/test/java/com/blackship/battlesheep/game/state/board/BoardModifierTest.java:52]: Why not use loop?

File: server/src/test/java/com/blackship/battlesheep/game/state/board/BoardModifierTest.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        assertEquals(givenBoard.getPositions().get(givenShip.get(1)), new Position(givenShip.get(1), FieldState.TAKEN));
+        assertEquals(givenBoard.getPositions().get(givenShip.get(2)), new Position(givenShip.get(2), FieldState.TAKEN));
+        assertEquals(givenBoard.getPositions().get(givenShip.get(3)), new Position(givenShip.get(3), FieldState.TAKEN));
+    }
+
+    @Test(dataProvider = "boardSheepsData")
+    public void insertShipsExpectBoardContainingShips(List<Ship> givenShips) {
+        //given StartingBoard
```
Comment (mhbratek) [inline server/src/test/java/com/blackship/battlesheep/game/state/board/BoardModifierTest.java:60]: I think it is not needed.

**Review by mhbratek — APPROVED** [review]
Think about ```StartingGame``` you have class```BoardModifier```

File: common/src/main/java/com/blackship/battlesheep/game/Position.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+    private final Integer position;
+    private FieldState fieldState;
+
+    public Position(Integer position, FieldState fieldState) {
+        this.position = position;
+        this.fieldState = fieldState;
+    }
+
+    public Position(Integer position) {
```
Comment (arturAR) [inline common/src/main/java/com/blackship/battlesheep/game/Position.java:17]: Maybe factory method? ;)

File: server/src/main/java/com/blackship/battlesheep/game/state/board/BoardModifier.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+        Map<Integer, Position> boardPositions = new HashMap<>(board.getPositions());
+        shipPositions.forEach(data -> boardPositions.replace(data, new Position(data, FieldState.TAKEN)));
+        return new GameBoard(boardPositions);
+    }
+
+    public static Board insertShips(List<Ship> ships) {
+        Map<Integer, Position> insertShips = new HashMap<>();
+        for (Ship ship : ships) {
+            for (Integer i : ship.getPositions()) {
+                insertShips.put(i, new Position(i, FieldState.TAKEN));
```
Comment (arturAR) [inline server/src/main/java/com/blackship/battlesheep/game/state/board/BoardModifier.java:31]: Maybe change method insertShip and use it?

File: server/src/test/java/com/blackship/battlesheep/game/state/board/BoardModifierTest.java [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+    @DataProvider
+    private Object[][] boardSheepData() {
+        return new Object[][] {
+                {new ArrayList<>(Arrays.asList(1, 11, 12, 13))},
+                {new ArrayList<>(Arrays.asList(1, 11 , 21, 31))}
+        };
+    }
+
+    @DataProvider
+    private Object[][] boardSheepsData() {
```
Comment (arturAR) [inline server/src/test/java/com/blackship/battlesheep/game/state/board/BoardModifierTest.java:34]: Sheeps? Not ships? 
6738204-shaun-the-sheep-the-movie-643-643

**Review by arturAR — APPROVED** [review]

**Pull request merged**